### PR TITLE
Revert "Disable Net5CompatFileStream by default (#50166)"

### PR DESCRIPTION
**Pull request by carlossanlop**
Partially addresses https://github.com/dotnet/runtime/issues/51141

Unblocks https://github.com/dotnet/sdk/pull/16810

This reverts commit 8b2ad5bccff29715d924d0c2d191bc978aaaf320 to unblock dotnet/sdk.

### CODE REVIEW

**Review by jozkee — APPROVED** [review]

**Review by adamsitnik — APPROVED** [review]
@carlossanlop thank you for doing this! I hope to have a bug fix tomorrow

**Review by jeffhandley — APPROVED** [review]
Thanks, @carlossanlop. Out of curiosity, could we alternatively enable the compat switch in the SDK project(s) to unblock that build? Or do you think there are too many projects affected by the bug right now?

**Comment by carlossanlop** [comment]
> Thanks, @carlossanlop. Out of curiosity, could we alternatively enable the compat switch in the SDK project(s) to unblock that build? Or do you think there are too many projects affected by the bug right now?

I am unsure of the impact on other repos. We think it's safer at this point having this change reverted, do our tests again, make sure the issue is reproduced locally, and get it fixed before we turn on the compat switch by default again.

**Comment by jeffhandley** [comment]
Makes sense; thanks!

**Comment by adamsitnik** [comment]
I have a fix, please don't merge this PR ;)

**Comment by adamsitnik** [comment]
Fix of the actual bug: https://github.com/dotnet/runtime/pull/51151

**Comment by carlossanlop** [comment]
Thanks Adam! If the CI in your PR finishes successfully, we will merge that. Otherwise, we'll merge this PR.

**Pull request closed without merging**
